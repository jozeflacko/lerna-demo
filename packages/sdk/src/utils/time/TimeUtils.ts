export default class TimeUtils {

    private static getDate(): Date {
        return new Date();
    }

    public static getToday(): string {
        const date: Date = this.getDate();
        const [day, month, year] = [date.getDate(), date.getMonth(), date.getFullYear()];
        return day + "." + month + "." + year;
    }

    public static getCurrentTime(): string {
        const date: Date = this.getDate();
        const [hours, minutes, seconds] = [date.getHours(), date.getMinutes(), date.getSeconds()];
        return hours + ":" + minutes + ":" + seconds;
    }
}