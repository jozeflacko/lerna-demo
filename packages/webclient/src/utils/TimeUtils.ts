import * as sdk from 'sdk';

export default class TimeUtils {
    static getToday() {
        return sdk.utils.TimeUtils.getCurrentTime() + " " + sdk.utils.TimeUtils.getToday();
    }
}