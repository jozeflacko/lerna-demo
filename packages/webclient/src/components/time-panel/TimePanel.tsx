import * as React from "react";
import TimeUtils from '../../utils/TimeUtils';

interface Props {
    prefixText: string;
}

interface State {
    today: string
}

export default class App extends React.Component<Props,State> {
    
    readonly state: State = {
        today: ""
    }

    constructor(props: Props) {
        super(props);

        this.state = {
            today: TimeUtils.getToday()
        }
    }

    render() {
        const  {today} = this.state ;
        
        return (
            <div>
                <span>{this.props.prefixText}</span>
                <span>{today}</span>
            </div>
        );
    }
}