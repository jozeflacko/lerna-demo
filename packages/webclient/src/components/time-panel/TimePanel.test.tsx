import TimePanel from './TimePanel';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import TimeUtils from '../../utils/TimeUtils';

// mock TimeUtils
jest.mock('../../utils/TimeUtils');
const mockedTimeUtils = TimeUtils as jest.Mocked<typeof TimeUtils>;
mockedTimeUtils.getToday.mockReturnValue("22:22:22 22.2.2022");

describe("Testing Main App Component", ()=>{


    const renderer = TestRenderer.create(<TimePanel prefixText="Hello"/>);
    const instance = renderer.getInstance();

    it.only("App will match snapshot",()=> {
        expect(renderer.toJSON()).toMatchSnapshot();
    });    

});
