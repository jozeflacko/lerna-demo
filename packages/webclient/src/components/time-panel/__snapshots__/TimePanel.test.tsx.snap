// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Main App Component App will match snapshot 1`] = `
<div>
  <span>
    Hello
  </span>
  <span>
    22:22:22 22.2.2022
  </span>
</div>
`;
